src/regtech_api_commons/api/dependencies.py

```diff
@@ -0,0 +1,81 @@
+import os
+from http import HTTPStatus
+from typing import List
+
+import httpx
+from fastapi import Request
+from starlette.authentication import AuthCredentials
+
+from regtech_api_commons.api.exceptions import RegTechHttpException
+from regtech_api_commons.models.auth import AuthenticatedUser
+
+
+def verify_lei(inst_api_url: str):
+    def lei_active_check(request: Request, lei: str) -> None:
+        res = httpx.get(inst_api_url + lei, headers={"authorization": request.headers["authorization"]})
+        lei_obj = res.json()
+        if not lei_obj["is_active"]:
+            raise RegTechHttpException(
+                status_code=HTTPStatus.FORBIDDEN, name="Request Forbidden", detail=f"LEI {lei} is in an inactive state."
+            )
+
+    return lei_active_check
+
+
+def verify_user_lei_relation(request: Request, *args, **kwargs) -> None:
+    if lei := kwargs.get("lei"):
+        user: AuthenticatedUser = request.user
+        auth: AuthCredentials = request.auth
+        if not is_admin(auth):
+            if lei not in user.institutions:
```

Comment:
I had to add a check for authentication in https://github.com/cfpb/sbl-filing-api/issues/135 because unauthenticated calls to the dependency were causing an error to be thrown up (and resulting in 500s). Should that be done here?

Reply:
ah yeah... this triggers before the requires authenticated if I remember correctly... yeap, I'll add that.